File: Program/Clothing.java
/* 
    Saya Tattha Maharany Yasmin Akbar dengan NIM 2201805 mengerjakan soal Latihan 2 
    dalam Praktikum mata kuliah Desain dan Pemrograman Berbasis Objek, untuk keberkahan-Nya
    maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

// Clothing merupakan child dari shirt
class Clothing extends Shirt
{
    // atribut private yang dibutuhkan
    private String size;
    private String material;
    private char gender;

    // konstruktor
    Clothing() {

    }
    Clothing(String size, String material, char gender)
    {
        this.size = size;
        this.material = material;
        this.gender = gender;
    }
    
    // getter dan setter
    public String get_size()
    {
        return this.size;
    }
    public void set_size(String size)
    {
        this.size = size;
    }
    public String get_material()
    {
        return this.material;
    }
    public void set_material(String material)
    {
        this.material = material;
    }
    public char get_gender()
    {
        return this.gender;
    }
    public void set_gender(char gender)
    {
        this.gender = gender;
    }
}

File: CPP/Program/Clothing.cpp
/* 
    Saya Tattha Maharany Yasmin Akbar dengan NIM 2201805 mengerjakan soal Latihan 2 
    dalam Praktikum mata kuliah Desain dan Pemrograman Berbasis Objek, untuk keberkahan-Nya
    maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

// library
#include <bits/stdc++.h>
#include "Shirt.cpp"

using namespace std;

class Clothing : public Shirt
{
    // attributes
    private:
        string size, material;
        char gender;

    public:
        Clothing(){

        }

    // construct
    Clothing(string size, string material, char gender)
    {
        this->size = size;
        this->material = material;
        this->gender = gender;
    }

    // getter dan setter
    string get_size()
    {
        return this->size;
    }
    void set_size(string size)
    {
        this->size = size;
    }
    string get_material()
    {
        return this->material;
    }
    void set_material(string material)
    {
        this->material = material;
    }
    char get_gender()
    {
        return this->gender;
    }
    void set_gender(char gender)
    {
        this->gender = gender;
    }

    ~Clothing(){}; // destruct
};
